crypto/mimc7/mimc7.go

```diff
@@ -0,0 +1,96 @@
+package mimc7
+
+import (
+	"encoding/hex"
+	"fmt"
+	"math/big"
+
+	"github.com/arnaucube/go-snark/bn128"
+	"github.com/arnaucube/go-snark/fields"
+	"github.com/ethereum/go-ethereum/crypto"
+)
+
+// MaxFieldVal is the maximum value that a element of a field can have, fitting inside an R Finite Field. This value is equal to 2**253 - 1
+const MaxFieldVal = "0x1fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff"
+const SEED = "iden3_mimc"
+
+// RElem is a big.Int of maximum 253 bits
+type RElem *big.Int
+
+//BigIntsToRElems converts from array of *big.Int to array of RElem
+func BigIntsToRElems(arr []*big.Int) ([]RElem, error) {
+	var o []RElem
+	maxFieldBytes, err := hex.DecodeString(MaxFieldVal[2:])
+	if err != nil {
+		return o, err
+	}
+	maxVal := new(big.Int).SetBytes(maxFieldBytes)
```

Comment:
maxFieldBytes and maxVal are constant values and yet they are computed for every conversion.  Could you move this outside of the function as global vars so that they are only computed once?

Reply:
done